test/test_convergence.jl

```diff
@@ -55,37 +68,54 @@ function check_over_dry_states(
                             thermo_params = SFP.thermodynamics_params(param_set)
-                            ts_sfc_test = Thermodynamics.PhaseDry{FT}(psfc.e_int, psfc.ρ)
-                            ts_int_test = Thermodynamics.PhaseDry{FT}(pint.e_int, pint.ρ)
-                            state_in = SF.InteriorValues(pint.z, (pint.u / 10, pint.v / 10), ts_int_test)
-                            state_sfc = SF.SurfaceValues(FT(0), (FT(0), FT(0)), ts_sfc_test)
-                            sc = SF.ValuesOnly{FT}(; state_in, state_sfc, z0m, z0b)
-                            DSEᵥ_int = TD.virtual_dry_static_energy(thermo_params, ts_int_test, pint.e_pot)
-                            DSEᵥ_sfc = TD.virtual_dry_static_energy(thermo_params, ts_sfc_test, psfc.e_pot)
-                            ΔDSEᵥ = DSEᵥ_int - DSEᵥ_sfc
-                            if abs(ΔDSEᵥ) <= FT(SFP.cp_d(param_set) / 10)
+                            ts_sfc = Thermodynamics.PhaseDry{FT}(p_sfc.e_int, p_sfc.ρ)
+                            ts_int = Thermodynamics.PhaseDry{FT}(p_int.e_int, p_int.ρ)
+                            sc = assemble_surface_conditions(p_int, p_sfc, ts_int, ts_sfc, z0m, z0b)
+                            ΔDSEᵥ = compute_dse_diff(thermo_params, ts_int, ts_sfc, p_int, p_sfc)
+                            if abs(ΔDSEᵥ) <= tol_neutral
                                 counter[3] += 1
                             else
                                 sign(ΔDSEᵥ) == 1 ? counter[1] += 1 : counter[2] += 1
                             end
-                            @test try
-                                sfcc =
-                                    SF.surface_conditions(param_set, sc, sch; maxiter, soltype = RS.VerboseSolution())
-                                true
-                            catch
-                                false
-                            end
+                            #@test try
+                                sfcc = SF.surface_conditions(
+                                        param_set,
+                                        sc,
+                                        sch;
+                                        maxiter,
+                                        soltype = RS.VerboseSolution(),
+                                        noniterative_stable_sol = gryanik_noniterative,
+                                )
+                                if abs(ΔDSEᵥ) <= tol_neutral && gryanik_noniterative == false
+                                    @test isinf(sfcc.L_MO)
+                                else    
+                                if sign(sfcc.L_MO) != sign(ΔDSEᵥ)
+                                  @show ΔDSEᵥ
+                                  @show sfcc
+                                end
+                                    @test sign.(sfcc.ρτxz) == -sign(p_int.u)
+                                    @test sign.(sfcc.ρτyz) == -sign(p_int.v)
+                                #   @test sign(sfcc.L_MO) == sign(ΔDSEᵥ)
```

Comment:
is this failing?

Reply:
Sorted - was done for local checks / debugging. Resolved now.